packages/blockchain-wallet-v4-frontend/src/data/analytics/reducers.js

```diff
@@ -0,0 +1,41 @@
+import * as AT from './actionTypes'
+import { insert, path, contains, assoc } from 'ramda'
+import assert from 'assert'
+import { TRACKING_ACTIONS } from './model'
+import { SHOW_CONFIRMATION } from '../components/exchange/actionTypes'
+import moment from 'moment'
+
+const INITIAL_STATE = {
+  swapTiming: null,
+  events: []
+}
+
+const analytics = (state = INITIAL_STATE, action) => {
+  const { type, payload } = action
+
+  switch (type) {
+    case AT.LOG_EVENT: {
+      assert(path(['trackingData', 'length'], payload), 'trackingData cannot be empty')
```

Comment:
`assert` module will throw in case of failure, right? Don't think it's good idea to throw error inside reducer

Reply:
also why adding a whole extra module to the production?

Reply:
+1 im also not a fan of assert, need to think too much, the whole rest of the code base throws Errors

Reply:
Sure, there are other ways to handle malformed payloads we can use here. BTW- assert is already used in the core, so it wasn't added just for this case.

Reply:
nice, core usage is cool, reminds me that sometimes developers install postgres to use it as csv :D